sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -1627,3 +1628,108 @@ def sdc_pandas_dataframe_groupby_impl(self, by=None, axis=0, level=None, as_inde
         return init_dataframe_groupby(self, column_id, grouped, sort)
 
     return sdc_pandas_dataframe_groupby_impl
+
+
+def df_add_column_codelines(self, key):
+    """Generate code lines for DF add_column"""
+    func_lines = []
+    if self.columns:
+        func_lines += df_length_codelines(self)
+        func_lines += [
+            '  if length == 0:',
+            '    raise SDCLimitation("Could not set item for DataFrame with empty columns")',
+            '  elif length != len(value):',
+            '    raise ValueError("Length of values does not match length of index")',
+        ]
+    else:
+        func_lines += ['  length = len(value)']
+    func_lines += df_index_codelines(self)  # provide res_index = ...
```

Comment:
So the lines df_index_codelines produces actually use length defined above, that's not very good. 
Do you think it would be better if it's impl is changed to take str variable length as a param and use where needed it if it's not None?